log/gps/gps_mgl_enigma_waypoint.ksy

```diff
@@ -0,0 +1,148 @@
+    Exceptions:
+      MGL Avionics reserves copyright and denies use of this data format for any military or related purpose.
+
+doc: |
+  Waypoint format used by Enigma series of EFIS instruments from MGL Avionics.
+  It is common to create an index file containing a simple list of indexes, one to each entry in the waypoint file, sorted by a particular criteria (for example distance of the waypoint to a given location), rather than sorting the actual waypoint list itself. This is a method commonly employed by database applications.
+
+doc-ref:
+  - https://mglavionics.co.za/Docs/Enigma%20Waypoint%20format.pdf
+
+seq:
+  - id: waypoints
+    type: waypoint
+    repeat: eos
+    doc: There is no demand that waypoints are stored in any particular order in the waypoint file. They might be sorted by short name or waypoint type but this is of little consequence. The order of waypoints in the file matters only for route files. Route waypoints are stored in the file in order of their appearance in the route. A route should have at least one waypoint.
+
+types:
+  waypoint:
+    seq:
+      - id: lat
+        type: sph_coord
+        doc: Positive values are in Nothern Hemisphere
+      - id: lon
+        type: sph_coord
+        doc: Positive values are in Eastern Hemisphere
+      - id: data_raw
+        type: u4
+      - id: type
+        type: u1
+        enum: type
+      - id: name_short
+        type: dc_str(6)
+        doc: This entry is used as “key” and in aviation terms would contain the short identifier of an airfield or navaid beacon. This key is used to lookup related data in other databases, for example an airport data base containing details of an airport such as frequencies and runway headings.
+      - id: name_long
+        type: dc_str(27)
+        doc: |
+          The long name contains a human readable description of the waypoint.
+          For example: “Los Angeles International”. The long name is arbitrary and not used as search key for any purpose.
+
+    instances:
+      is_data_frequency:
+        value: 9 <= type.to_i and type.to_i <= 25
+      altitude:
+        value: "( ( data_raw < 0x80000000 ) ? data_raw : ( ( data_raw & 0x7FFFFFFF ) - 0x80000000 ) )"
+        doc: in feet
+        if: not is_data_frequency
+      frequency:
+        value: data_raw
+        doc: in kHz
+        if: is_data_frequency
+
+  dc_str:
+    params:
+      - id: max_size
+        type: u1
+    seq:
+      - id: size
+        type: u1
+      - id: value
+        type: str
+        size: value_size
+      - id: garbage
+        type: str
+        size: max_size - value_size
+    instances:
+      value_size:
+        value: max_size > size ? size : max_size
+
+  sph_coord:
+    seq:
+      - id: raw
+        type: s4
+    instances:
+      value:
+        value: raw / 180000
```

Comment:
This looks like it should result in a floating-point number instead:

```suggestion
        value: raw / 180000.0
```

Reply:
Yeah, I haven't tested it at all. Only checked that it compiles.